File: transcoder/src/main/java/com/vsnt/transcoder/config/Secrets.java
package com.vsnt.transcoder.config;

public class Secrets {
    public static final String AWS_SECRET_KEY = System.getenv("AWS_SECRET_KEY") ;
    public static final String AWS_ACCESS_KEY_ID = System.getenv("AWS_ACCESS_KEY_ID") ;
    public static final String AWS_RAW_BUCKET_NAME = System.getenv("AWS_RAW_BUCKET_NAME") ;
    public static final String AWS_TRANSCODED_BUCKET_NAME = System.getenv("AWS_TRANSCODED_BUCKET_NAME") ;
    public static final String DOCKER_TRANSCODER_CONTAINER_IMAGE = System.getenv("DOCKER_TRANSCODER_CONTAINER_IMAGE") ;
    public static final String CLOUD_FRONT_URL = System.getenv("CLOUD_FRONT_URL") ;
}

File: rtmp_server/src/secrets.js
// config/secrets.js

const Secrets = {
    AWS_SECRET_KEY:                    process.env.AWS_SECRET_KEY,
    AWS_ACCESS_KEY_ID:                 process.env.AWS_ACCESS_KEY_ID,
    AWS_RAW_BUCKET_NAME:               process.env.AWS_RAW_BUCKET_NAME,
    AWS_TRANSCODED_BUCKET_NAME:        process.env.AWS_TRANSCODED_BUCKET_NAME,
    DOCKER_TRANSCODER_CONTAINER_IMAGE: process.env.DOCKER_TRANSCODER_CONTAINER_IMAGE,
    CLOUD_FRONT_URL:                   process.env.CLOUD_FRONT_URL,
};

// // Validate all secrets are present on startup
// const missing = Object.entries(Secrets)
//     .filter(([_, value]) => !value)
//     .map(([key]) => key);

// if (missing.length > 0) {
//     throw new Error(`Missing required environment variables: ${missing.join(', ')}`);
// }

module.exports = Secrets;